File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/interpreter/lab/multiply.cpp
// multiply.cpp

#include <iostream>
#include "binary.h"
#include "dp_exception.h"
class expression;
#include "multiply.h"

multiply::multiply(expression * a_operand1, expression * a_operand2)
: binary(a_operand1, a_operand2)
{
}

void multiply::output(std::ostream & a_ostream) const
{
	output_infix(a_ostream, "*");
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/interpreter/lab/Multiply.java
// Multiply.java

import java.io.PrintWriter;

public class Multiply extends Binary
{
	public Multiply(Expression aOperand1, Expression aOperand2)
	{
		super(aOperand1, aOperand2);
	}

	public void output(PrintWriter aPrintWriter)
	{
		outputInfix(aPrintWriter, "*");
	}
}
